### PR TITLE
[ISSUE #20] add boltdb support for businessstore、platformstore and circuitbreakerstore

### PR DESCRIPTION
**Pull request by chuntaojun**
achive boltdb store ability

1. BusinessStore
2. PlatformStore
3. CircuitbreakerStore

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (andrewshan) [inline .vscode/launch.json]: don't commit the temporary workspace files

File: polaris-server.yaml [hunk 2]
```diff
@@ -72,10 +72,10 @@ naming:
   # 健康检查
```
Comment (andrewshan) [inline polaris-server.yaml]: you shouldn't change the default configuration

File: store/boltdbStore/business.go [hunk 3]
```diff
@@ -18,52 +18,217 @@
...
+	"sync"
 	"time"
+
+	"github.com/polarismesh/polaris-server/common/log"
+	"github.com/polarismesh/polaris-server/common/model"
+	"github.com/polarismesh/polaris-server/store"
+)
+
+const (
+	DataTypeBusiness string = "business"
```
Comment (andrewshan) [inline store/boltdbStore/business.go:32]: change to tblNameBusiness

File: store/boltdbStore/business.go [hunk 4]
```diff
@@ -18,52 +18,217 @@
...
 )
 
 type businessStore struct {
 	handler BoltHandler
+	lock    *sync.RWMutex
```
Comment (andrewshan) [inline store/boltdbStore/business.go:37]: mutex is useless here

File: store/boltdbStore/business.go [hunk 5]
```diff
@@ -18,52 +18,217 @@
...
 	return nil
 }
 
-// 更新业务集
-func (b *businessStore) UpdateBusiness(business *model.Business) error {
-	//TODO
+// UpdateBusiness 更新业务集
+func (bs *businessStore) UpdateBusiness(b *model.Business) error {
+	if b.ID == "" || b.Name == "" || b.Owner == "" {
+		log.Errorf("[Store][database] update business missing some params: %+v", b)
```
Comment (andrewshan) [inline store/boltdbStore/business.go:97]: [Store][database] change to [Store][boltdb]

File: store/boltdbStore/business.go [hunk 6]
```diff
@@ -18,52 +18,217 @@
...
+
+// listAllBusiness 列出所有的 Business 信息
+func (bs *businessStore) listAllBusiness() ([]*model.Business, error) {
+	lock := bs.lock
+	lock.RLock()
+	defer lock.RUnlock()
+
+	dbOp := bs.handler
+
+	result, err := dbOp.LoadValuesByFilter(DataTypeBusiness, []string{}, &model.Business{}, func(m map[string]interface{}) bool {
```
Comment (andrewshan) [inline store/boltdbStore/business.go:219]: LoadValuesByFilter change to LoadValuesAll

File: store/boltdbStore/business.go [hunk 7]
```diff
@@ -18,52 +18,217 @@
...
+	}
+
 	return nil, nil
-}
\ No newline at end of file
+}
+
+// GetMoreBusiness 根据mtime获取增量数据
+func (bs *businessStore) GetMoreBusiness(mtime time.Time) ([]*model.Business, error) {
+	result, err := bs.listAllBusiness()
```
Comment (andrewshan) [inline store/boltdbStore/business.go:192]: you shouldn't load all values here and filter it, you can use the LoadValuesByFilter to do the filtering

File: store/boltdbStore/business.go [hunk 8]
```diff
@@ -18,52 +18,217 @@
...
-	//TODO
-	return nil, nil
+// ListBusiness 查询owner下业务集
+func (bs *businessStore) ListBusiness(owner string) ([]*model.Business, error) {
+	if owner == "" {
+		log.Errorf("[Store][business] list business missing owner")
+		return nil, errors.New("List Business Mising param owner")
+	}
+
+	result, err := bs.listAllBusiness()
```
Comment (andrewshan) [inline store/boltdbStore/business.go:141]: you shouldn't load all values here and filter it, you can use the LoadValuesByFilter to do the filtering

File: store/boltdbStore/business.go [hunk 9]
```diff
@@ -18,52 +18,217 @@
...
-	return nil
+// UpdateBusinessToken 更新业务集token
+func (bs *businessStore) UpdateBusinessToken(bid string, token string) error {
+	if bid == "" || token == "" {
+		log.Errorf("[Store][business] update business token missing some params")
+		return errors.New("Update Business Token missing some params")
+	}
+
+	// first: get business by bid
+	b, err := bs.GetBusinessByID(bid)
```
Comment (andrewshan) [inline store/boltdbStore/business.go:125]: useless you shouldn't call it

File: .vscode/launch.json (hunk 1 above)
Comment (chuntaojun) [reply .vscode/launch.json]: already remove

File: store/boltdbStore/business.go (hunk 9 above)
Comment (chuntaojun) [reply store/boltdbStore/business.go:125]: already fix

File: store/boltdbStore/business.go (hunk 8 above)
Comment (chuntaojun) [reply store/boltdbStore/business.go:141]: already fix

File: store/boltdbStore/business.go (hunk 7 above)
Comment (chuntaojun) [reply store/boltdbStore/business.go:192]: already fix

File: store/boltdbStore/business.go (hunk 6 above)
Comment (chuntaojun) [reply store/boltdbStore/business.go:219]: already fix

File: common/utils/funcs.go [hunk 10]
```diff
@@ -88,3 +89,28 @@ func CreateInstanceModel(serviceID string, req *api.Instance) *model.Instance {
 	instance.Proto = protoIns
 	return instance
 }
+
+func ConvertFilter(filters map[string]string) map[string][]string {
```
Comment (andrewshan) [inline common/utils/funcs.go:93]: 外部导出的函数需要有注释

File: common/utils/funcs.go [hunk 11]
```diff
@@ -88,3 +89,28 @@ func CreateInstanceModel(serviceID string, req *api.Instance) *model.Instance {
...
+	for k, v := range filters {
+		val := make([]string, 0)
+		val = append(val, v)
+		newFilters[k] = val
+	}
+
+	return newFilters
+}
+
+func CollectFilterFields(filters map[string]string) []string {
```
Comment (andrewshan) [inline common/utils/funcs.go:105]: 外部导出的函数需要有注释

File: store/boltdbStore/business.go [hunk 12]
```diff
@@ -18,52 +18,189 @@
...
+	}
+	return ans, nil
+}
+
+// listAllBusiness 列出所有的 Business 信息
+func (bs *businessStore) listAllBusiness() ([]*model.Business, error) {
+
+	dbOp := bs.handler
+
+	result, err := dbOp.LoadValues(tblBusiness, []string{}, &model.Business{})
```
Comment (andrewshan) [inline store/boltdbStore/business.go:193]: 这里用LoadValuesAll方法

**Pull request merged**